File: backend/src/main/java/com/project2/domain/post/dto/comment/ListCommentResponseDTO.java
package com.project2.domain.post.dto.comment;

import lombok.Getter;

import java.util.ArrayList;
import java.util.List;

@Getter
public class ListCommentResponseDTO {
    private final Long id;
    private final String content;
    private final String nickname;
    private final Long parentId;
    private final List<ListCommentResponseDTO> children = new ArrayList<>();

    public ListCommentResponseDTO(Long id, String content, String nickname, Long parentId) {
        this.id = id;
        this.content = content;
        this.nickname = nickname;
        this.parentId = parentId;
    }
}

File: backend/src/main/kotlin/com/project3/domain/post/dto/comment/ListCommentResponseDTO.kt
package com.project3.domain.post.dto.comment

class ListCommentResponseDTO(
    val id: Long,
    val content: String,
    val nickname: String,
    val parentId: Long?,
    val children: MutableList<ListCommentResponseDTO> = mutableListOf()
) {
    constructor(id: Long, content: String, nickname: String, parentId: Long?) : this(
        id, content, nickname, parentId, mutableListOf()
    )
}